sale_order_action_invoice_create_hook/hooks.py

```diff
@@ -0,0 +1,148 @@
+                            order.client_order_ref
+                    invoices[group_key].write(vals)
+                if line.qty_to_invoice > 0 or \
+                        (line.qty_to_invoice < 0 and final):
+                    if pending_section:
+                        pending_section.invoice_line_create(
+                            invoices[group_key].id,
+                            pending_section.qty_to_invoice
+                        )
+                        pending_section = None
+                    line.invoice_line_create(invoices[group_key].id,
+                                             line.qty_to_invoice)
+                    # START HOOK
+                    # Change to true if new lines are added
+                    new_lines = True
+                    # END HOOK
+                if references.get(invoices.get(group_key)):
+                    if order not in references[invoices[group_key]]:
+                        references[invoice] = references[invoice] | order
+
+        # START HOOK
+        # WAS: if not invoices:
+        # Check if new lines have been added in order to determine whether
+        # there are invoice lines or not
+        if not new_lines and not self.env.context.get('no_check_lines', False):
+            raise UserError(_('There is no invoicable line.'))
+        # END HOOK
+
+        for invoice in invoices.values():
+            invoice.compute_taxes()
+            if not invoice.invoice_line_ids:
+                raise UserError(_('There is no invoicable line.'))
+            # If invoice is negative, do a refund invoice instead
+            if invoice.amount_untaxed < 0:
+                invoice.type = 'out_refund'
+                for line in invoice.invoice_line_ids:
+                    line.quantity = -line.quantity
+            # Use additional field helper function (for account extensions)
+            for line in invoice.invoice_line_ids:
+                line._set_additional_fields(invoice)
+            # Necessary to force computation of taxes. In account_invoice,
+            # they are triggered by onchanges, which are not triggered when
+            # doing a create.
+            invoice.compute_taxes()
+            # Idem for partner
+            so_payment_term_id = invoice.payment_term_id.id
+            invoice._onchange_partner_id()
+            # To keep the payment terms set on the SO
+            invoice.payment_term_id = so_payment_term_id
+            invoice.message_post_with_view(
+                'mail.message_origin_link',
+                values={'self': invoice, 'origin': references[invoice]},
+                subtype_id=self.env.ref('mail.mt_note').id)
+        return [inv.id for inv in invoices.values()]
+
+    if not hasattr(SaleOrder, 'action_invoice_create_original'):
+        SaleOrder.action_invoice_create_original = \
+            SaleOrder.action_invoice_create
+
+    SaleOrder.action_invoice_create = new_action_invoice_create
```

Comment:
This should be done in `_register_hook` method and not here.

Reply:
Ok like this?